File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestPercent.js
print("getPercent")
if(typeof getPercent !== 'undefined')
	throw "getPercent already defined!"
getPercent = callback("getPercent")
getPercentType = typeof getPercent
print(getPercentType)
print(getPercent)
if(getPercentType !== 'number')
	throw "getPercent not number!"
callback("setPercent",getPercent)

print("getPercentVector")
if(typeof getPercentVector !== 'undefined')
	throw "getPercentVector already defined!"
getPercentVector = callback("getPercentVector")
getPercentVectorType = typeof getPercentVector[0]
print(getPercentVectorType)
print(getPercentVector)
if(getPercentVectorType !== 'number')
	throw "getPercentVector not number!"
callback("setPercentVector",getPercentVector)

print("getPercentVectorAsList")
if(typeof getPercentVectorAsList !== 'undefined')
	throw "getPercentVectorAsList already defined!"
getPercentVectorAsList = callback("getPercentVectorAsList")
getPercentVectorAsListType = typeof getPercentVectorAsList[0]
print(getPercentVectorAsListType)
print(getPercentVectorAsList)
if(getPercentVectorAsListType !== 'number')
	throw "getPercentVectorAsList not number!"
callback("setPercentVectorAsList",getPercentVectorAsList)

print("getPercentMatrix")
if(typeof getPercentMatrix !== 'undefined')
	throw "getPercentMatrix already defined!"
getPercentMatrix = callback("getPercentMatrix")
getPercentMatrixType = typeof getPercentMatrix[0][0]
print(getPercentMatrixType)
print(getPercentMatrix)
if(getPercentMatrixType !== 'number')
	throw "getPercentMatrix not number!"
callback("setPercentMatrix",getPercentMatrix)

print("getPercentMatrixAsList")
if(typeof getPercentMatrixAsList !== 'undefined')
	throw "getPercentMatrixAsList already defined!"
getPercentMatrixAsList = callback("getPercentMatrixAsList")
getPercentMatrixAsListType = typeof getPercentMatrixAsList[0][0]
print(getPercentMatrixAsListType)
print(getPercentMatrixAsList)
if(getPercentMatrixAsListType !== 'number')
	throw "getPercentMatrixAsList not number!"
callback("setPercentMatrixAsList",getPercentMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/callback/ParametersAndReturnsTestPercent.rb
puts("getPercent")
unless (defined?(getPercent)).nil? || getPercent.nil?
	raise "getPercent already defined!"
end
getPercent = callback("getPercent")
getPercentType = getPercent.class
puts(getPercentType)
puts(getPercent)
if(getPercentType != Float)
	raise "getPercent not Float!"
end
callback("setPercent", getPercent)

puts("getPercentVector")
unless (defined?(getPercentVector)).nil? || getPercentVector.nil?
	raise "getPercentVector already defined!"
end
getPercentVector = callback("getPercentVector")
getPercentVectorType = getPercentVector[0].class
puts(getPercentVectorType)
puts(getPercentVector)
if(getPercentVectorType != Float)
	raise "getPercentVector not Float!"
end
callback("setPercentVector", getPercentVector)

puts("getPercentVectorAsList")
unless (defined?(getPercentVectorAsList)).nil? || getPercentVectorAsList.nil?
	raise "getPercentVectorAsList already defined!"
end
getPercentVectorAsList = callback("getPercentVectorAsList")
getPercentVectorAsListType = getPercentVectorAsList[0].class
puts(getPercentVectorAsListType)
puts(getPercentVectorAsList)
if(getPercentVectorAsListType != Float)
	raise "getPercentVectorAsList not Float!"
end
callback("setPercentVectorAsList", getPercentVectorAsList)

puts("getPercentMatrix")
unless (defined?(getPercentMatrix)).nil? || getPercentMatrix.nil?
	raise "getPercentMatrix already defined!"
end
getPercentMatrix = callback("getPercentMatrix")
getPercentMatrixType = getPercentMatrix[0][0].class
puts(getPercentMatrixType)
puts(getPercentMatrix)
if(getPercentMatrixType != Float)
	raise "getPercentMatrix not Float!"
end
callback("setPercentMatrix", getPercentMatrix)

puts("getPercentMatrixAsList")
unless (defined?(getPercentMatrixAsList)).nil? || getPercentMatrixAsList.nil?
	raise "getPercentMatrixAsList already defined!"
end
getPercentMatrixAsList = callback("getPercentMatrixAsList")
getPercentMatrixAsListType = getPercentMatrixAsList[0][0].class
puts(getPercentMatrixAsListType)
puts(getPercentMatrixAsList)
if(getPercentMatrixAsListType != Float)
	raise "getPercentMatrixAsList not Float!"
end
callback("setPercentMatrixAsList", getPercentMatrixAsList)
